### PR TITLE
[Pointcloud] Adding busses for easier better editor UX

### LINKED ISSUE #53: [Pointcloud] Add some useful Editor Buses
Upon clustering the scene point cloud I have currently 50 separate point clouds to work with (and this is only a beginning). I would like to have some functionalities provided by the Editor version of the Component that will make working with point clouds more pleasant.

- [x] AzFramework::BoundsRequestBus for localizing the Aabb would be great and it will make the Z shortcut work.
- [x] AzToolsFramework::EditorComponentSelectionRequestsBus would also be nice for click and drag selection of particular point cloud.
- [ ] ViewportDebugDisplay has a lower priority, but it should be cheap once we have Bounds implemented.

### PR DESCRIPTION
**Pull request by Fireronin**
Resolves #53 

Right now first two are implemented  
AzFramework::BoundsRequestBus
AzToolsFramework::EditorComponentSelectionRequestsBus

### CODE REVIEW

**Comment by michalpelka** [comment]
Please give a meaningful name to PR.

File: Gems/Pointcloud/Assets/Shaders/Pointclouds/Pointclouds.shader [hunk 1]
```diff
@@ -8,8 +8,8 @@
   },
   "GlobalTargetBlendState" : {
       "Enable" : false,
-      "BlendSource" : "Zero",
-      "BlendDest" : "One",
+      "BlendSource" : "AlphaSource",
```
Comment (michalpelka) [inline Gems/Pointcloud/Assets/Shaders/Pointclouds/Pointclouds.shader:11]: Is this change required by this feature?

Comment (Fireronin) [reply Gems/Pointcloud/Assets/Shaders/Pointclouds/Pointclouds.shader:11]: No, I'm not sure how it got there

**Comment by Fireronin** [comment]
I did a bit of testing and cleaned up changed files

**Review by zakmat — APPROVED** [review]
Yesterday I worked with the code a little since there were some issues with the version I reviewed. I have reapplied some changes on top of yours. A final solution is satisfactory. Thanks a lot!

**Review by michalpelka — CHANGES_REQUESTED** [review]
Code cannot be compiled with 2409.
In 2409 API for :
```
        AZ::Aabb GetWorldBounds() const override;
        AZ::Aabb GetLocalBounds() const override;
```
Some methods have are const . In 2310 not.

**Review by michalpelka — APPROVED** [review]

**Comment by michalpelka** [comment]
I  think that is not a blocker to merge. Those gems should work with release version of o3de that currently is 2309.
